pkg/common/volume_operations.go

```diff
@@ -112,23 +119,37 @@ func (vo *VolumeOperationsImpl) CreateVolume(ctx context.Context, v api.Volume)
 		)
 
 		if util.IsStorageClassLVG(sc) {
-			requiredBytes = AlignSizeByPE(requiredBytes)
+			requiredBytes = capacityplanner.AlignSizeByPE(requiredBytes)
 		}
 
-		ac = vo.acProvider.SearchAC(ctxWithID, v.NodeId, requiredBytes, v.StorageClass)
+		capReader := capacityplanner.NewACReader(vo.k8sClient, vo.log, true)
+		resReader := capacityplanner.NewACRReader(vo.k8sClient, vo.log, true)
+
+		capacityManager := vo.createCapacityManager(v, capReader, resReader)
+		plan, err := capacityManager.PlanVolumesPlacing(ctxWithID, []*api.Volume{&v})
+		if err != nil {
+			ll.Errorf("error while planning placing for volume: %s", err.Error())
+			return nil, err
+		}
+		noResourceErr := status.Errorf(
+			codes.ResourceExhausted, "there is no suitable drive for volume %s", v.Id)
+		if plan == nil {
+			return nil, noResourceErr
+		}
+		if v.NodeId == "" {
```

Comment:
In general case if we don't find AC on SelectedNode we should try another one. Could you please leave todo in that place with such info.

Reply:
In "plan" we store all available options for volume placing.  SelectNode() will return node which 100% has required capacity.
All checks in plan.GetACForVolume required when volume has preferred node